props/blaster.h

```diff
@@ -260,9 +268,28 @@ class Blaster : public PROP_INHERIT_PREFIX PropBase {
     }
   }
 
+  RefPtr<BufferedWavPlayer> auto_player_;
+
   void Loop() override {
     PropBase::Loop();
     PollNextAction();
+    if (auto_firing_ && GetWavPlayerPlaying(&SFX_auto)) {
```

Comment:
You're still running GetWavPlayerPlaying() every loop.
Also, the logic here seems a bit wonky. It ought to be something like:

if (auto_firing_) {
  if (!auto_plaeyr_) {
    auto_player_ = GetWavPlayerPlaying(&SFX_auto);
  }
  if (auto_player_) {
      // do stuff
  }
} else {
  auto_player_.Reset();
}

Reply:
Note that the reset is important, otherwise it will keep the same player between auto-firing sessions.

Reply:
So can I move that block into Fire() in the MODE_AUTO part? I don't know why I'm in Loop. I guess leftover from incorrect State Mchine logic.
Trying that, but Reset() makes it unhappy:
`class RefPtr<BufferedWavPlayer>' has no member named 'Reset'

Reply:
I think I meant Free(), not Reset().

Reply:
Some of this needs to be in loop() to happen over and over again.
However, setting and unsetting the auto_player_ might be better to put elsewhere.

Reply:
ok thanks. Seems the problem keeps happening if i move the wav length detection out of loop.
It doesn't pickup the global auto_player_ inside loop. 
I get NULL POINTER when checking elapsed time against length.
Submitted changes but not working yet.

Reply:
Leaning toward making a new `Function()`, doing
`auto_player_ = GetWavPlayerPlaying(&SFX_auto)` in there, 
then in the loop, get the length with
`1000 * Function()->length()`
but what type is auto_player?
That would need a `return *type*` yes?

Reply:
@profezzorn The best I can determine with limited understanding is that in order to remove `auto_player_ = GetWavPlayerPlaying()` out of loop, auto_player_ needs to be a global variable or pointer. I'm not sure where that gets done, and if it's making a new function to do so, if it gets an * or $ or... beyond my level on comprehension at the moment.
I DO know that the code works if it's all in loop. 
Just trying to understand the issue with that ..."too expensive" I guess I get that it's too many calls for something that can be had once elsewhere.
But I'll need specific help on this as I've exhausted my ability at the moment.

Reply:
Confirming I can't get auto_player_ to be recognized inside Loop() if it's declared outside.
I'm sure it's something simple I just don't have a grasp on yet.

Reply:
Ok Progress...sort of.
Using &auto_player_, the RefPtr works inside Loop().
However, the auto.wav is played stretched out, but not like 1/2 samplerate, just time stretched to what sounds like 2x duration to me.
Any idea what I did?
Committed.